### PR TITLE
cleanup PR_BRANCH definition

### PR DESCRIPTION
**Pull request by ataylorme**
See #72

### CODE REVIEW

File: .circleci/test-example-repo.sh [hunk 1]
```diff
@@ -4,8 +4,25 @@ set -ex
...
+# If we are on the master branch
+if [[ $CIRCLE_BRANCH == "master" ]]
+then
+    PR_BRANCH="dev-master"
+else
+    # If this is a pull request use the PR number
+    if [[ -z "$CIRCLE_PULL_REQUEST" ]]
+    then
+        # Stash PR number
+        PR_NUMBER=${CIRCLE_PULL_REQUEST##*/}
```
Comment (greg-1-anderson) [inline .circleci/test-example-repo.sh]: Looks like this is not evaluating correctly.

**Comment by greg-1-anderson** [comment]
Looks good to me. Once the minor errors in the variable assignments are fixed up and the tests are passing here we can merge and see how it does on the master branch.

File: .circleci/test-example-repo.sh (hunk 1 above)
Comment (ataylorme) [inline .circleci/test-example-repo.sh]: Odd, I re-ran the build with SSH and just tested this section and it was working properly.

`CIRCLE_PULL_REQUEST` return the pull request URL, `PR_NUMBER=${CIRCLE_PULL_REQUEST##*/}` just strips everything up to and including the last dash which leaves the pull request number.

screen shot 2017-07-19 at 10 47 46 am

Should `BUILD_TOOLS_VERSION` just return `1.x` or `2.x`?

Comment (greg-1-anderson) [inline .circleci/test-example-repo.sh]: I'm not sure what is causing the discrepancy here.

As for the result, ultimately BUILD_TOOLS_VERSION is used to select a release via Composer, so it should be `dev-foo` for a PR on the `foo` branch, or `dev-master` for the master branch, or `^1` or `~1` or `1.x` for the 1.x branch.

Comment (ataylorme) [inline .circleci/test-example-repo.sh]: Thanks for the clarification

File: .circleci/test-example-repo.sh [hunk 2]
```diff
@@ -4,12 +4,37 @@ set -ex
...
+        PR_BRANCH="dev-$CIRCLE_BUILD_NUM"
+    fi
+fi
+
 SOURCE_COMPOSER_PROJECT="$1"
 TARGET_REPO=$GITHUB_USERNAME/$TERMINUS_SITE
 TARGET_REPO_WORKING_COPY=$HOME/$TERMINUS_SITE
-BUILD_TOOLS_VERSION=${PR_BRANCH:-$CIRCLE_BRANCH}
+
+# If we are on the 1.x branch set the build tools version to 1.x
```
Comment (ataylorme) [inline .circleci/test-example-repo.sh]: @greg-1-anderson does this work for you?

**Comment by ataylorme** [comment]
Not sure why the [WordPress test](https://circleci.com/gh/pantheon-systems/terminus-build-tools-plugin/357) is failing on `terminus build:env:create -n "$TERMINUS_SITE.dev" "$TERMINUS_ENV" --yes --notify="$NOTIFY"`. Any thoughts @greg-1-anderson?

**Comment by greg-1-anderson** [comment]
Maybe it's because [we're using the same site name](https://github.com/pantheon-systems/terminus-build-tools-plugin/blob/90439f2d8cd84a013fdd5e67f12a224f67895c80/.circleci/test-example-repo.sh#L5) for all of our tests?

**Comment by ataylorme** [comment]
> Maybe it's because we're using the same site name for all of our tests?

Agreed! I think Drupal starts first which is why it is passing.

**Comment by stevector** [comment]
> Maybe it's because we're using the same site name for all of our tests?

I don't think so. Drupal and WordPress are running in different build numbers.

**Comment by stevector** [comment]
WordPress failed on a Terminus 504.

https://circleci.com/gh/pantheon-ci-bot/build-tools-357/1

```
 Server error: `GET https://terminus.pantheon.io/api/sites/37e690e1-230f-47da-8cb3-4fd5aa6220f1/workflows/5f8e9656-6ca2-11e7-bf82-bc764e1022a9` resulted in a `504 Gateway Time-out` response:
<html>

<head><title>504 Gateway Time-out</title></head>

<body bgcolor="white">

<center><h1>504 Gateway Time-out</h1>< (truncated...)
 

```

**Comment by greg-1-anderson** [comment]
There weren't any tests running, so I selected and deleted all of the stale Pantheon sites. Hopefully I did that before Steve restarted the WordPress test - or at least before the test created its Pantheon site.

**Comment by greg-1-anderson** [comment]
OK, the deletes are done, and the site create for the test is still in progress, so hopefully this test run will pass.

**Comment by ataylorme** [comment]
@greg-1-anderson and @stevector I think the issue is:

`composer create-project -n -d ~/.terminus/plugins pantheon-systems/terminus-build-tools-plugin:^1`

The passing D8 test uses
`composer create-project -n -d ~/.terminus/plugins pantheon-systems/terminus-build-tools-plugin:$BUILD_TOOLS_VERSION --stability=dev`

**Comment by ataylorme** [comment]
Yup, need to update [`circle.yml` in `example-wordpress-composer`](https://github.com/pantheon-systems/example-wordpress-composer/blob/2028db1764ca2d884ece4673e3885f62842f5dc3/circle.yml#L60)

**Comment by greg-1-anderson** [comment]
Yeah, I'm not sure that the 1.x branch would work here. Best to make that change for consistency in any event.

**Comment by ataylorme** [comment]
Well, it did¯\_(ツ)_/¯

**Comment by ataylorme** [comment]
The change to build tools version is in https://github.com/pantheon-systems/example-wordpress-composer/pull/15

File: .circleci/config.yml [hunk 3]
```diff
@@ -39,7 +39,7 @@ jobs:
             - run:
                 name: Test WordPress
                 # @todo, change to pantheon-systems
-                command: ./.circleci/test-example-repo.sh "pantheon-systems/example-wordpress-composer:dev-master"
+                command: ./.circleci/test-example-repo.sh "pantheon-systems/example-wordpress-composer:dev-dynamic-build-tools-version"
```
Comment (greg-1-anderson) [inline .circleci/config.yml]: n.b. `dev-test-2.x` in the example-drops-8-composer version (per my comment https://github.com/pantheon-systems/example-wordpress-composer/pull/15#issuecomment-316473855)

**Comment by ataylorme** [comment]
Seems Packagist isn't picking up on the `dynamic-build-tools-version` branch. Can you take a look @greg-1-anderson?

**Comment by greg-1-anderson** [comment]
@stevector Maybe https://packagist.org/packages/pantheon-systems/example-wordpress-composer is not configured to auto-update? I am not an admin on this project, so I cannot tell, but the branch @ataylorme mentions above is not known to Packagist, and I suspect that is the reason why.

**Comment by stevector** [comment]
Yes, I forgot to configure the autoupdate. It is on now.

**Comment by ataylorme** [comment]
`TERM environment variable not set`

Looks like a dependency of WordHat is expecting to be run in a Terminal.

**Comment by greg-1-anderson** [comment]
A lot of Linux commands will complain like that if TERM is not set. I see that message in other CI tests; it usually does not interfere with anything.

If it is being printed too often and interfering with your test output, you could add `export TERM=dumb` at the beginning of the problematic script.

**Comment by ataylorme** [comment]
> you could add export TERM=dumb at the beginning of the problematic script.

@greg-1-anderson adding `TERM` to the `run-behat` script didn't work but adding it as an environment variable to CircleCI via `.circleci/config.yml` did.

See [this CircleCI forum thread](https://discuss.circleci.com/t/circleci-terminal-is-a-tty-but-term-is-not-set/9965/3) for others having the issue as well.

Now it's hung up on asking for the codeserver password. Perhaps something is wrong with the SSH key?

**Comment by ataylorme** [comment]
@greg-1-anderson WordPress is still failing even after SSH was cleared up. `terminus build-env:merge -n "$TERMINUS_SITE.$TERMINUS_ENV" --yes` exists with `1`. Ran out of time and likely won't be able to come back to this until the 31st as I have WordCamp Boston tomorrow and this weekend then two trainings and a QBR next week.

If you or @stevector can pick this up that would be lovely.

**Comment by greg-1-anderson** [comment]
It looks like this is the `ours` vs. `theirs` vs just-force-push-it idea you had for build:env:merge.

screen shot 2017-07-20 at 4 36 33 pm

**Comment by greg-1-anderson** [comment]
Actually, the `master` branch (and the code in this PR) is currently using strategy `ours`. Based on the screenshot above, it looks like the wrong version of the Terminus Build Tools Plugin is being used in the test site that is created.

**Comment by greg-1-anderson** [comment]
GREEN!

**Pull request merged**